script-test/playbackstrategies/modifiers/html5tests.js

```diff
@@ -0,0 +1,3260 @@
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+              player.beginPlaybackFrom(0);
+              player.pause();
+
+              metaDataCallback({});
+
+              mockVideoMediaElement.play.calls.reset();
+
+              player.resume();
+
+              expect(mockVideoMediaElement.play).toHaveBeenCalledTimes(1);
+            });
+
+            it(' Play Not Called On Media Element When Resume In Buffering State Before Metadata', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+              player.beginPlaybackFrom(0);
+              player.pause();
+
+              mockVideoMediaElement.play.calls.reset();
+
+              player.resume();
+
+              expect(mockVideoMediaElement.play).not.toHaveBeenCalled();
+            });
+
+            it(' Pause Passed Through To Media Element When In Buffered State', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+              player.beginPlaybackFrom(0);
+              player.pause();
+
+              metaDataCallback({});
+
+              expect(mockVideoMediaElement.pause).toHaveBeenCalledTimes(1);
+            });
+
+            it(' Load Called On Media Element When Initialise Media Is Called', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+
+              expect(mockVideoMediaElement.load).toHaveBeenCalledTimes(1);
+            });
+
+            it(' Media Element Preload Attribute Is Set To Auto', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+
+              expect(mockVideoMediaElement.preload).toEqual('auto');
+            });
+
+            it(' Play From Sets Current Time And Calls Play On Media Element When In Playing State', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+              player.beginPlaybackFrom(0);
+
+              metaDataCallback({});
+              finishedBufferingCallback();
+
+              mockVideoMediaElement.play.calls.reset();
+
+              player.playFrom(10);
+
+              expect(mockVideoMediaElement.play).toHaveBeenCalledTimes(1);
+              expect(mockVideoMediaElement.currentTime).toEqual(10);
+            });
+
+            it(' Get Player Element Returns Video Element For Video', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+
+              expect(player.getPlayerElement()).toEqual(mockVideoMediaElement);
+            });
+
+            it(' Get Player Element Returns Audio Element For Audio', functi
```

Comment:
This and the next few tests seem to be doing too much. The point of the these tests is that we don't actually perform a seek, and our player code feigns a transition from buffering and then back to playing, as if we had seeked. But these tests emulate real buffering instead.